server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -351,6 +352,41 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
         return new SortFieldAndFormat(field, fieldType.docValueFormat(null, null));
     }
 
+    public boolean canRewriteToMatchNone() {
+        return nestedSort == null && (missing == null || "_last".equals(missing));
+    }
+
+    /**
+     * Returns whether all values of the given {@link QueryShardContext#getIndexReader()} are within the
+     * primary sort value provided in the <code>rawBottomSortValues</code>.
+     */
+    public boolean isBottomSortWithinShard(QueryShardContext context, Object[] rawBottomSortValues) {
```

Comment:
Is there a reason why we pass have pass an array of bottom sort values, but we use only 1st element? Should we pass only the 1st element  – `Object[] rawBottomSortValue`?